### PR TITLE
Explain how to publish a new finder

### PR DESCRIPTION
**Pull request by Davidslv**
This commit explains the steps needed to publish a new finder.

Trello: https://trello.com/c/X0I3cfJS/335-write-documentation-about-how-to-create-a-finder

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,11 +55,37 @@ The application has jasmine tests, which can be accessed at `/specs` when the ap
...
-1. If required, add a schema to [alphagov/rummager](http://github.com/alphagov/rummager) describing your document type -- eg
-   [https://github.com/alphagov/rummager/blob/master/config/schema/default/doctypes/cma_case.json](https://github.com/alphagov/rummager/blob/master/config/schema/default/doctypes/cma_case.json)
+1. If required, add a schema to [alphagov/rummager](http://github.com/alphagov/rummager) describing your document type -- [example](https://github.com/alphagov/rummager/blob/master/config/schema/document_types/cma_case.json)
 2. Publish a Finder Content Item to the content store. See the doc for [Finder Content Item](https://github.com/alphagov/finder-frontend/blob/master/docs/finder-content-item.md) for more info.
 3. Ensure your documents are indexed in [alphagov/rummager](http://github.com/alphagov/rummager) correctly.
 
+
+### How to publish a finder?
+
+#### When the publishing app is whitehall
```
Comment (tijmenb) [inline README.md]: What would you think of moving this chapter to whitehall? If for example the rake task changes name this repo can get out of date quickly.

File: README.md [hunk 2]
```diff
@@ -55,11 +55,37 @@ The application has jasmine tests, which can be accessed at `/specs` when the ap
...
+
+- You will need to create a JSON file in the whitehall repository, under [whitehall/lib/finders][finders-folder]
+- You can base it on one of the existing files in that folder
+- Double-check the filter format and document noun - the filter format is used for rummager to return the data, while the document noun is displayed to the user
```
Comment (MatMoore) [inline README.md]: Maybe we could have a screenshot here with the various bits labelled?

File: README.md (hunk 1 above)
Comment (MatMoore) [inline README.md]: It seems like the publishing app is kind of arbitrary for finders. Should we include a sentence explaining this?

Like "if your finder contains only whitehall documents use whitehall; if it's specialist documents, the publishing app is specialist publisher..."?

File: README.md [hunk 3]
```diff
@@ -55,11 +55,37 @@ The application has jasmine tests, which can be accessed at `/specs` when the ap
...
+- The default_documents_per_page key can be used to paginate very long finders (see [whitehall/lib/finders/case_studies.json][case-studies] for an example)
+- Running the [finders:publish rake task][rake-task] will publish your new finder to the [publishing-api](https://github.com/alphagov/publishing-api), the the route defined in the JSON will be taken over by finder-frontend
+
+[finders-folder]: https://github.com/alphagov/whitehall/tree/master/lib/finders
+[case-studies]: https://github.com/alphagov/whitehall/blob/master/lib/finders/case_studies.json
+[rake-task]: https://github.com/alphagov/whitehall/blob/master/lib/tasks/publish_finders.rake
+
+#### How can I know the format needed?
```
Comment (tijmenb) [inline README.md]: Let's link to the GitHub source, it's good enough and won't change. I'm not sure about using separate GitHub pages for separate projects. The current rummager docs look crappy.

**Comment by Davidslv** [comment]
I've changed the documentation, moved the information about Whitehall into https://github.com/alphagov/whitehall/pull/2865 

I will squash the commits when everyone's happy with the changes.

File: README.md [hunk 4]
```diff
@@ -59,32 +59,19 @@ The application has jasmine tests, which can be accessed at `/specs` when the ap
...
-- You can base it on one of the existing files in that folder
-- Double-check the filter format and document noun - the filter format is used for rummager to return the data, while the document noun is displayed to the user
-- The default_documents_per_page key can be used to paginate very long finders (see [whitehall/lib/finders/case_studies.json][case-studies] for an example)
-- Running the [finders:publish rake task][rake-task] will publish your new finder to the [publishing-api](https://github.com/alphagov/publishing-api), the the route defined in the JSON will be taken over by finder-frontend
+You can double check the filter by performing the following search using rummager:
 
-[finders-folder]: https://github.com/alphagov/whitehall/tree/master/lib/finders
-[case-studies]: https://github.com/alphagov/whitehall/blob/master/lib/finders/case_studies.json
-[rake-task]: https://github.com/alphagov/whitehall/blob/master/lib/tasks/publish_finders.rake
+http://rummager.dev.gov.uk/search.json?filter_NAME=_FORMAT_NAME_
```
Comment (MatMoore) [inline README.md]: this still refers to format, maybe `filter_NAME=VALUE`?

File: README.md [hunk 5]
```diff
@@ -59,32 +59,19 @@ The application has jasmine tests, which can be accessed at `/specs` when the ap
 2. Publish a Finder Content Item to the content store. See the doc for [Finder Content Item](https://github.com/alphagov/finder-frontend/blob/master/docs/finder-content-item.md) for more info.
 3. Ensure your documents are indexed in [alphagov/rummager](http://github.com/alphagov/rummager) correctly.
 
+### How can I know the filter needed?
```
Comment (MatMoore) [inline README.md]: this title still sounds a bit strange to me - how about calling it something like "How to add a fixed filter"?

**Comment by MatMoore** [comment]
I added a couple more more inline comments but looks good to me.

I was thinking it would also be helpful to link to @tijmenb's new schema docs for finders: https://alphagov.github.io/govuk-developers/content-schemas/finder/

**Comment by tijmenb** [comment]
Those schema docs are very prototype, not ready for prod yet. Hopefully we can get something permanent in December.

File: README.md (hunk 4 above)
Comment (Davidslv) [inline README.md]: 👍

File: README.md (hunk 5 above)
Comment (Davidslv) [inline README.md]: will changed it 👍

**Comment by Davidslv** [comment]
@MatMoore & @tijmenb I've addressed your comments, and I've rebased the branch.

Thank you

**Pull request merged**